mantis-common/src/main/java/io/mantisrx/runtime/descriptor/DeploymentStrategy.java

```diff
@@ -16,34 +16,50 @@
 
 package io.mantisrx.runtime.descriptor;
 
+import io.mantisrx.shaded.com.fasterxml.jackson.annotation.JsonInclude;
+import io.mantisrx.shaded.com.fasterxml.jackson.annotation.JsonInclude.Include;
 import io.mantisrx.shaded.com.fasterxml.jackson.annotation.JsonProperty;
 import java.util.Map;
 import lombok.Builder;
 import lombok.EqualsAndHashCode;
+import lombok.Getter;
 import lombok.Singular;
 import lombok.ToString;
 
 @Builder
 @EqualsAndHashCode
 @ToString
 public class DeploymentStrategy {
-    @Singular(ignoreNullCollections = true) private Map<Integer, StageDeploymentStrategy> stages;
+    @Singular(ignoreNullCollections = true, value = "stage")
+    @Getter
+    private final Map<Integer, StageDeploymentStrategy> stageDeploymentStrategyMap;
+
+    /**
+     * If this field is not empty, it's used to indicate this is a resource cluster stack deployment and will be hosted
+     * to the given resource cluster Id.
+     */
+    @Getter
+    @JsonInclude(Include.NON_NULL)
+    private final String resourceClusterId;
```

Comment:
Also, can we conform to ClusterID type here instead of String?

Reply:
Actually, I realized this is the runtime contract that doesn't have dep on the "clusterID" in the mantis control plane. Given the fact that this is to be used to set the label value underlying the strong type doesn't help much (since it's going to be converted to label string again).